projects/components/src/quick-search/quick-search.service.ts

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { Injectable, OnDestroy } from '@angular/core';
+import { QuickSearchProvider } from './quick-search.provider';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class QuickSearchService {
+    registrations: QuickSearchProvider[] = [];
+
+    /**
+     * Register a search provider
+     * @param provider The search provider {@link QuickSearchProvider}
+     */
+    public registerProvider(provider: QuickSearchProvider): void {
```

Comment:
It's because unregistering can kind of fail if the provider is not registered. Since you added unit tests to it, I just kept it working, it doesn't seem wrong